lib/identity/helpers/auth.rb

```diff
@@ -161,27 +161,96 @@ def perform_oauth_refresh_dance
 
         @cookie.access_token            = token["access_token"]["token"]
         @cookie.access_token_expires_at =
-          Time.now + token["access_token"]["expires_in"]
+          Time.now.getlocal + token["access_token"]["expires_in"]
 
         raise "missing=access_token"  unless @cookie.access_token
         raise "missing=expires_in"    unless @cookie.access_token_expires_at
 
         log :oauth_refresh_dance_complete, session_id: @cookie.session_id
       end
+    end
+
+    # Attempt to resolve an oauth/authorize request
+    def call_authorize(authorize_params = get_authorize_params)
+      # clear anything that might be left over in the session
+      @cookie.authorize_params = nil
```

Comment:
Unfortunately there's a chicken and egg problem here, the call to `authorize_params` below on line 187 (https://github.com/heroku/identity/pull/192/files#diff-31af98b21cf0db8ec15ce1602918f908R187) potentially needs `@cookie.authorize_params` intact.